src/record/import/utils/file.ts

```diff
@@ -4,6 +4,23 @@ import iconv from "iconv-lite";
 
 export type SupportedImportEncoding = "utf8" | "sjis";
 
+export const openFsStreamWithEncode: (
+  filePath: string,
+  encoding?: SupportedImportEncoding
+) => { stream: NodeJS.ReadWriteStream; format: string } = (
+  filePath,
+  encoding = "utf8"
+) => {
+  const format = extractFileFormat(filePath);
+  if (format === "json" && encoding !== "utf8") {
+    throw new Error("source file is JSON and JSON MUST be encoded with UTF-8");
+  }
+  const stream = fs
+    .createReadStream(filePath)
```

Comment:
@ueokande 
I found we can rewind Stream from the file descriptor.
But there are some risks.

1. We have to specify `autoClose: false` to `createReadStream` to prevent the file descriptor closing automatically after all data is read from Stream. \[[link](https://nodejs.org/api/fs.html#filehandlecreatereadstreamoptions)\]
If any code is missing `autoClose: false`, the program will crash on the next opening stream.
Also, we have to close file descriptor at the end of our program.
(Actually, cli-kintone is CLI and the fd will be closed at the end of the program by stack unwinding)

2. When we recreate the stream with `createReadStream`, the seek position in the file is kept from the last call.
So we have to specify `start: 0` to `createReadStream` to reset the position. \[[link](https://stackoverflow.com/questions/31606497/is-it-possible-to-rewind-a-file-descriptor-cursor-in-nodejs)\]


They are natural behaviors and as the same as primitive file access like a `fgetc()` of C.
But we have to mind resetting the position and not close fd when we use `createReadStream`.
In my opinion, it is a little annoying and leads to mistakes.
(Especially, the first one will become a trap.)

```javascript
const fd = await fs.promises.open(filePath);
console.log("fd: ", fd.fd);                 // fd: 27

// read all data from Stream
const stream1 = fd.createReadStream({ autoClose: false });
let count1 = 0;
for await (const data of stream1) count1++;
console.log("data read: ", count1); // data read:  10
console.log("fd: ", fd.fd);                 // fd: 27

// read all data from Stream again
const stream2 = fd.createReadStream({ start: 0, autoClose: false });
let count2 = 0;
for await (const data of stream2) count2++;
console.log("data read: ", count2); // data read:  10
console.log("fd: ", fd.fd);                  // fd: 27

// close file descriptor
await fd.close();
console.log("fd: ", fd.fd);                  // fd: -1
```

Reply:
I agree it is helpful practice and we can use it.
Let's consider it in the future!